napi_js_data.md

```diff
@@ -0,0 +1,873 @@
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API allocates a JavaScript object with external data attached to it.
+This is used to wrap native objects and project them into JavaScript.
+The API allows the caller to pass in a finalize callback, in case the 
+underlying native resource needs to be cleaned up when the wrapper 
+JavaScript object gets collected.
+
+### *napi_create_external_arraybuffer*
+
+#### Signature
+```C
+napi_status
+napi_create_external_arraybuffer(napi_env env,
+                                 void* external_data,
+                                 size_t byte_length,
+                                 napi_finalize finalize_cb,
+                                 void* finalize_hint,
+                                 napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  external_data`: Pointer to the underlying byte buffer of the 
+ArrayBuffer
+- `[in]  byte_length`: The length in bytes of the underlying buffer
+- `[in]  finalize_cb`: Optional callback to call when the ArrayBuffer is 
+being collected
+- `[in]  finalize_hint`: Optional hint to pass to the finalize callback 
+during collection
+- `[out] result`: A `napi_value` representing a JavaScript ArrayBuffer
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API returns an N-API value corresponding to a JavaScript ArrayBuffer.
+The underlying byte buffer of the ArrayBuffer is externally allocated and 
+managed. The caller must ensure that the byte buffer remains valid until the
+finalize callback is called.
+
+JavaScript ArrayBuffers are described in 
+[Section 24.1](https://tc39.github.io/ecma262/#sec-arraybuffer-objects) 
+of the ECMAScript Language Specification.
+
+### *napi_create_external_buffer*
+
+#### Signature
+```C
+napi_status napi_create_external_buffer(napi_env env,
+                                        size_t length,
+                                        void* data,
+                                        napi_finalize finalize_cb,
+                                        void* finalize_hint,
+                                        napi_value* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  length`: Size in bytes of the input buffer (should be the same as 
+the size of the new buffer)
+- `[in]  data`: Raw pointer to the underlying buffer to copy from
+- `[in]  finalize_cb`: Optional callback to call when the ArrayBuffer is
+being collected
+- `[in]  finalize_hint`: Optional hint to pass to the finalize callback
+during collection
+- `[out] result`: A `napi_value` representing a `node::Buffer`
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This API allocates a `node::Buffer` object and initializes it with data
+backed by the passed in buffer. It exists for compat with older Node versions.
```

Comment:
Please update this to the same text as the other `node::Buffer`-related APIs, i.e. the "compat with older Node versions" part.